drivers/modem/modem_cmd_handler.c

```diff
@@ -646,27 +645,46 @@ void modem_cmd_handler_tx_unlock(struct modem_cmd_handler *handler)
 }
 
 int modem_cmd_handler_init(struct modem_cmd_handler *handler,
-			   struct modem_cmd_handler_data *data)
+			   struct modem_cmd_handler_data *data,
+			   const struct modem_cmd_handler_setup *setup)
 {
-	if (!handler || !data) {
+	/* Verify arguments */
+	if (NULL == handler || NULL == data || NULL == setup) {
 		return -EINVAL;
 	}
 
-	if (!data->match_buf_len) {
+	/* Verify setup */
+	if (setup->match_buf == NULL || setup->match_buf_len == 0 || setup->buf_pool == NULL
+		|| (NULL != setup->response_cmds && 0 == setup->response_cmds_len)
```

Comment:
Normally the operator should be at the right?

```suggestion
		(NULL != setup->response_cmds && 0 == setup->response_cmds_len) ||
```

and not sure why we changed to yoda conditions from this line?